### PR TITLE
Redirect back to My Works after delete

### LINKED ISSUE #2447: When I delete a work from My Work Listing I get forwarded to the catalog
### Descriptive summary

In Sufia 7 when I delete a work from my work listing instead of staying on the work listing I get forwarded to the catalog.  This also occurs from my highlights, my collections and files shared with me.
### Expected behavior

I would expect that if I delete a work from a My Listing (Works, Highlights shares) that I would return to the My Listing after the deletion occurs.
### Actual behavior

What happens currently is that I get sent to the catalog index.

screen shot 2016-08-10 at 7 31 42 am
### Steps to reproduce the behavior

When I have works in the system 
And I go to My Work Listing
And I select delete from the Action Menu 
And I confirm the delete
Then I should return to the work listing
### Related work

Link to related tickets or prior related work here.

### PR DESCRIPTION
**Pull request by hortongn**
Fixes #2447

### CODE REVIEW

File: app/controllers/concerns/sufia/works_controller_behavior.rb [hunk 1]
```diff
@@ -72,6 +72,14 @@ def after_update_response
         end
       end
 
+      def after_destroy_response(title)
+        flash[:notice] = "Deleted #{title}"
+        respond_to do |wants|
+          wants.html { redirect_to sufia.dashboard_works_path }
```
Comment (jeremyf) [inline app/controllers/concerns/sufia/works_controller_behavior.rb]: As written, the `flash[:notice]` will be set even for JSON responses. You can move the `flash[:notice]` to `redirect_to` command via the following:

```ruby
wants.html { redirect_to sufia.dashboard_works_path, notice: "Deleted #{title}" }
```

File: spec/controllers/curation_concerns/generic_works_controller_spec.rb [hunk 2]
```diff
@@ -186,4 +186,15 @@
       end
     end
   end
+
+  describe "#delete" do
+    let!(:work) { create(:work, title: ['test title'], user: user) }
+
+    context "after deletion" do
+      it "redirects to My Works" do
```
Comment (jeremyf) [inline spec/controllers/curation_concerns/generic_works_controller_spec.rb]: Please add a spec for JSON format.

File: spec/controllers/curation_concerns/generic_works_controller_spec.rb [hunk 3]
```diff
@@ -186,4 +186,15 @@
...
+        delete :destroy, params: { id: work }
+        expect(response).to redirect_to(Sufia::Engine.routes.url_helpers.dashboard_works_path)
```
Comment (jeremyf) [inline spec/controllers/curation_concerns/generic_works_controller_spec.rb]: Do you want to verify that the flash has been set?

File: app/controllers/concerns/sufia/works_controller_behavior.rb (hunk 1 above)
Comment (hortongn) [inline app/controllers/concerns/sufia/works_controller_behavior.rb]: 👍 fixed

File: spec/controllers/curation_concerns/generic_works_controller_spec.rb (hunk 2 above)
Comment (hortongn) [inline spec/controllers/curation_concerns/generic_works_controller_spec.rb]: 👍 fixed

File: spec/controllers/curation_concerns/generic_works_controller_spec.rb (hunk 3 above)
Comment (hortongn) [inline spec/controllers/curation_concerns/generic_works_controller_spec.rb]: 👍 fixed

**Comment by jcoyne** [comment]
@cam156 this looks good to me. Does it meet your expectations as the author of the ticket?

**Comment by hortongn** [comment]
Do we still need a 👍  from @cam156 on this before merging?

**Pull request merged**